### PR TITLE
feat: add python bindings for the v2 reader/writer

### PR DESCRIPTION
**Pull request by westonpace**
Prerequisites:

 - [x] https://github.com/lancedb/lance/pull/1965

### CODE REVIEW

**Review by wjones127 — APPROVED** [review]

File: python/python/lance/file.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
+#  Copyright (c) 2023. Lance Developers
```
Comment (eddyxu) [inline python/python/lance/file.py:1]: 2024 :)

File: python/python/lance/file.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    def to_table(self) -> pa.Table:
+        """
+        Return the results as a pyarrow Table
+        """
+        return self.reader.read_all()
+
+
+class LanceFileReader:
+    """
+    A file reader for reading Lance files
```
Comment (eddyxu) [inline python/python/lance/file.py:60]: Is this interface mostly internal utility? The public main interface is still `pyarrow.Dataset`?

File: python/python/tests/test_file.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+#  Copyright (c) 2023. Lance Developers
```
Comment (eddyxu) [inline python/python/tests/test_file.py:1]: 2024

File: python/python/lance/file.py (hunk 2 above)
Comment (westonpace) [reply python/python/lance/file.py:60]: Yes.  This is mainly for benchmarking / advanced use cases.

**Pull request merged**